### ISSUE TITLE
Create Classification & Regression head child classes

### ISSUE THREAD

**Issue by jakehlee** [emit-sds/SpecTf#26]
Wrapper classes that can take in any "base" class (e.g. encoder, decoder w/ cross attention, both) and attach a classification or regression head would be useful for different applications.

**Comment by jakehlee** [comment]
This was originally motivated by not wanting to overload `SpecTfEncoder()` with both classification and regression, instead creating `SpecTfClassifier()` and `SpecTfRegressor()` definitions. However, these two fingerprints would be identical.

Classifier models in PyTorch do not include `softmax` because the `nn.CrossEntropyLoss()` definition already includes it. For regression, some of our recent work includes multiple regression values to estimate uncertainty. Therefore, both classification and regression models need to support >=1 output dimensions without any activation layers afterwards.

`SpecTfEncoder()` (instead of just `SpecTf()`) also keeps it distinct from upcoming `SpecTfDecoder` (including cross-attention) and `SpecTfEncDec` (including the full Transformer formulation) models.

We will move forward by simply renaming the `n_classes` parameter to `dim_output`, and we'll take another look at this as collisions occur.

**Comment by jakehlee** [comment]
Closed in `dev` by #32
